model/buildlogger.go

```diff
@@ -563,3 +563,102 @@ func (l *Logs) Merge(ctx context.Context) (LogIterator, error) {
 
 	return NewMergingIterator(iterators...), nil
 }
+
+// TODO: Remove the following once task log migration is complete (EVG-13831).
+const (
+	// Task log types.
+	AgentLog  = "agent_log"
+	TaskLog   = "task_log"
+	SystemLog = "system_log"
+)
+
+// FindAndUpdateOutdatedTaskLogs finds evergreen task logs in the database
+// with the log type (agent, system, or task) in the info.proc_name field and
+// updates those documents to have the log type in the tags array.
+func FindAndUpdateOutdatedTaskLogs(ctx context.Context, env cedar.Environment) error {
+	if env == nil {
+		return errors.New("cannot find and update outdated task logs with a nil env")
+	}
+
+	logs, err := findOutdatedTaskLogs(ctx, env)
+	if err != nil {
+		return err
+	}
+
+	var (
+		agentLogs  []string
+		taskLogs   []string
+		systemLogs []string
+	)
+	for _, log := range logs {
+		switch log.Info.ProcessName {
+		case AgentLog:
+			agentLogs = append(agentLogs, log.ID)
+		case TaskLog:
+			taskLogs = append(taskLogs, log.ID)
+		case SystemLog:
+			systemLogs = append(systemLogs, log.ID)
+		}
+	}
+
+	if err = updateOutdatedTaskLogs(ctx, env, agentLogs, AgentLog); err != nil {
+		return err
+	}
+	if err = updateOutdatedTaskLogs(ctx, env, taskLogs, TaskLog); err != nil {
+		return err
+	}
+	return updateOutdatedTaskLogs(ctx, env, systemLogs, SystemLog)
+}
+
+func findOutdatedTaskLogs(ctx context.Context, env cedar.Environment) ([]Log, error) {
+	conf := &CedarConfig{}
+	conf.Setup(env)
+	if err := conf.Find(); err != nil {
+		return nil, errors.WithStack(err)
+	}
+	limit := conf.LogMigrationLimit
+	if limit <= 0 {
+		// Default limit to 1000.
+		limit = 1000
```

Comment:
Did you test that 1000 is an acceptably performant default to update once per minute?

Can we make it so that there's no default if it's unset? Just in case we want to turn off the migration.

Reply:
Yes, the I ran the `find` query with the limit set to 1000 multiple times and it ran on the order of 10s of ms. It does an index scan so it is extremely efficient.